engine/src/native-layer-mac.mm

```diff
@@ -157,7 +157,16 @@
 
 void MCNativeLayerMac::doSetViewportGeometry(const MCRectangle &p_rect)
 {
-    doSetGeometry(m_object->getrect());
+    MCRectangle t_rect;
+    t_rect = m_object->getrect();
+    
+    if (m_object -> gettype() == CT_PLAYER)
+    {
+        MCPlayer *t_player = (MCPlayer *)m_object;
+        t_rect = t_player -> getvideorect(t_rect);
+    }
+    
+    doSetGeometry(t_rect);
```

Comment:
@livecodepanos it's not a good idea to have code in MCNativeLayerMac that's specific to another control class. The bug arises from the player using a native layer that doesn't cover its entire rect. Perhaps we need a mechanism for getting the rect for a specific native layer from the control that it belongs to. Will need to give this some further thought.

Reply:
A simple fix may be to replace the original `doSetGeometry(m_object->getrect())` with `doSetGeometry(m_rect)`